File: src-tauri/src/snappit_res.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SnappitOcrEngine {
    Vision,
    Tesseract,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SnappitOcrResult {
    pub value: String,
    pub ocr: SnappitOcrEngine,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "kind", content = "payload", rename_all = "snake_case")]
pub enum SnappitResponse {
    Qr(Option<String>),
    Ocr(SnappitOcrResult),
}

File: shared/tauri/snappit_res.ts
import { ColorInfo } from "@/shared/tauri/screen_capture_api";

export interface SnappitQrResponse {
  kind: "qr";
  payload: string;
}

export type SnappitOcrEngine = "vision" | "tesseract";

export interface SnappitOcrPayload {
  value: string;
  ocr: SnappitOcrEngine;
}

export interface SnappitOcrResponse {
  kind: "ocr";
  payload: SnappitOcrPayload;
}

export interface SnappitDropperResponse {
  kind: "dropper";
  payload: ColorInfo;
}

export type SnappitResponse = SnappitQrResponse | SnappitOcrResponse | SnappitDropperResponse;
